parsers/basic.c

```diff
@@ -31,7 +31,9 @@ typedef enum {
 	K_LABEL,
 	K_TYPE,
 	K_VARIABLE,
-	K_ENUM
+	K_ENUM,
+	K_NAMESPACE,
+	K_NAMESPACE_END
```

Comment:
Introducing `K_NAMESPACE_END` for representing a keyword is good.
However, introducing `K_NAMESPACE_END` is added to `BasicKind`. Adding `K_NAMESPACE_END` doesn't make sense. Coupling kinds and keywords is not a good idea (though many parsers do so.)
We must split the enum for keywords and that for kinds.